### PR TITLE
🌱 Create internal/check package for ClusterClass

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR creates and internal/check package for ClusterClass by moving some code from the main controllers/topology folder.

**Anything else you would like to add:**
@vincepri I investigated a little bit introducing a singleton so we can have `check.Compatibility()..`, but at the end I rolled back those changes because it seemed to me to over-engineer the code given that at this stage we have only one kind of checks.
We can always iterate in the future if the need arises.

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -23,6 +23,7 @@ spec:
         - "--leader-elect"
         - "--metrics-bind-addr=localhost:8080"
         - "--feature-gates=MachinePool=${EXP_MACHINE_POOL:=false},ClusterResourceSet=${EXP_CLUSTER_RESOURCE_SET:=false},ClusterTopology=${CLUSTER_TOPOLOGY:=false}"
+        - "-v=3"
```
Comment (sbueringer) [inline config/manager/manager.yaml:26]: Is this an intended change?

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Comment by vincepri** [comment]
/lgtm cancel

Pending @sbueringer comment

File: config/manager/manager.yaml (hunk 1 above)
Comment (fabriziopandini) [reply config/manager/manager.yaml]: Testing left over, sorry. 
Removed

**Pull request merged**